File: python/markets.py
"""
Markets Example

List all available markets and HIP-3 DEXes.

No endpoint or private key needed — uses public API.
"""

from hyperliquid_sdk import HyperliquidSDK

# No endpoint or private key needed for read-only public queries
sdk = HyperliquidSDK()

# Get all markets
markets = sdk.markets()
print(f"Perp markets: {len(markets.get('perps', []))}")
print(f"Spot markets: {len(markets.get('spot', []))}")

# Show first 5 perp markets
print("\nFirst 5 perp markets:")
for m in markets.get("perps", [])[:5]:
    print(f"  {m['name']}: szDecimals={m.get('szDecimals')}")

# Get HIP-3 DEXes
dexes = sdk.dexes()
print(f"\nHIP-3 DEXes: {len(dexes)}")
for dex in dexes[:5]:
    print(f"  {dex.get('name', 'N/A')}")

File: typescript/markets.ts
#!/usr/bin/env npx ts-node
/**
 * Markets Example
 *
 * List all available markets and HIP-3 DEXes.
 *
 * No endpoint or private key needed — uses public API.
 */

import { HyperliquidSDK } from 'hyperliquid-sdk';

async function main() {
  // No endpoint or private key needed for read-only public queries
  const sdk = new HyperliquidSDK();

  // Get all markets
  const markets = await sdk.markets() as Record<string, unknown>;
  console.log(`Perp markets: ${((markets.perps || []) as unknown[]).length}`);
  console.log(`Spot markets: ${((markets.spot || []) as unknown[]).length}`);

  // Show first 5 perp markets
  console.log("\nFirst 5 perp markets:");
  for (const m of ((markets.perps || []) as unknown[]).slice(0, 5)) {
    const market = m as Record<string, unknown>;
    console.log(`  ${market.name}: szDecimals=${market.szDecimals}`);
  }

  // Get HIP-3 DEXes
  const dexesResponse = await sdk.dexes();
  const dexesList = Array.isArray(dexesResponse)
    ? dexesResponse
    : Object.values(dexesResponse);
  console.log(`\nHIP-3 DEXes: ${dexesList.length}`);
  for (const dex of dexesList.slice(0, 5)) {
    console.log(`  ${(dex as Record<string, unknown>).name || JSON.stringify(dex)}`);
  }
}

main().catch(console.error);
